packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts

```diff
@@ -76,6 +76,7 @@ export abstract class BaseDeployExecutor extends SfdxCommandletExecutor<
       this.logMetric(execution.command.logName, startTime, properties);
 
       try {
+        BaseDeployExecutor.errorCollection.clear();
         if (stdOut) {
```

Comment:
do we want to clear the diagnostics even if we don't parse the deploy results (when there is no stdOut) ?

Reply:
jk this makes sense to me